pkg/kubernetes/statefulset.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"github.com/billiford/go-clouddriver/pkg/kubernetes/manifest"
+	v1 "k8s.io/api/apps/v1"
+)
+
+type StatefulSet interface {
+	Object() *v1.StatefulSet
+	SetReplicas(*int32)
+	Status() manifest.Status
+}
+
+type statefulSet struct {
+	ss *v1.StatefulSet
+}
+
+func NewStatefulSet(m map[string]interface{}) StatefulSet {
+	s := &v1.StatefulSet{}
+	b, _ := json.Marshal(m)
+	_ = json.Unmarshal(b, &s)
+
+	return &statefulSet{ss: s}
+}
+
+func (ss *statefulSet) Object() *v1.StatefulSet {
+	return ss.ss
+}
+
+func (ss *statefulSet) SetReplicas(replicas *int32) {
+	ss.ss.Spec.Replicas = replicas
+}
+
+func (ss *statefulSet) Status() manifest.Status {
+	s := manifest.DefaultStatus
+	x := ss.ss
+
+	updStrategyType := x.Spec.UpdateStrategy.Type
+	if updStrategyType == "OnDelete" {
```

Comment:
use `strings.EqualFold` instead of `==` to match the [source](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesStatefulSetHandler.java#L118).

Reply:
Original Java code snippet:
  if (statefulSet.getSpec().getUpdateStrategy().getType().equalsIgnoreCase("ondelete")) {
      return Status.defaultStatus();
    }

I have converted to below Go code:
        s := manifest.DefaultStatus
        x := ss.ss

        updStrategyType := x.Spec.UpdateStrategy.Type
        if strings.EqualFold(updStrategyType, "OnDelete") {
                return s
        }


I am getting below error now:
# github.com/billiford/go-clouddriver/pkg/kubernetes [github.com/billiford/go-clouddriver/pkg/kubernetes.test]
pkg/kubernetes/statefulset.go:41: cannot use updStrategyType (type "k8s.io/api/apps/v1".StatefulSetUpdateStrategyType) as type string in argument to strings.EqualFold

Ginkgo ran 1 suite in 4.32626539s
Test Suite Failed


I have checked the conversion to the best of my ability. I could not resolve. Please assist.

Reply:
The underlying type of `StatefulSetUpdateStrategyType` is [a string](https://github.com/abursavich/kubernetes/blob/master/pkg/apis/apps/types.go#L74). I'll leave this up to you to figure out how to pass this to `strings.EqualFold` without getting the type error.